infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/InitializedState.java

```diff
@@ -33,19 +33,28 @@ public InitializedState(LogReplicationFSM logReplicationFSM) {
     public LogReplicationState processEvent(LogReplicationEvent event) throws IllegalTransitionException {
         switch (event.getType()) {
             case SNAPSHOT_SYNC_REQUEST:
-            case SNAPSHOT_WAIT_COMPLETE:
                 log.info("Start Snapshot Sync, requestId={}", event.getEventID());
                 // Set the id of the event that caused the transition to the new state
                 // This is used to correlate trim or error events that derive from this state
                 LogReplicationState snapshotSyncState = fsm.getStates().get(LogReplicationStateType.IN_SNAPSHOT_SYNC);
                 snapshotSyncState.setTransitionEventId(event.getEventID());
                 return snapshotSyncState;
-            case REPLICATION_START:
+            case SNAPSHOT_TRANSFER_COMPLETE:
+                log.info("Snapshot Sync transfer completed. Wait for snapshot apply to complete.");
+                WaitSnapshotApplyState waitSnapshotApplyState = (WaitSnapshotApplyState)fsm.getStates().get(LogReplicationStateType.WAIT_SNAPSHOT_APPLY);
+                waitSnapshotApplyState.setTransitionEventId(event.getEventID());
+                waitSnapshotApplyState.setBaseSnapshotTimestamp(fsm.getBaseSnapshot());
+                fsm.setBaseSnapshot(event.getMetadata().getLastTransferredBaseSnapshot());
+                fsm.setAckedTimestamp(event.getMetadata().getLastLogEntrySyncedTimestamp());
+                return waitSnapshotApplyState;
+            case LOG_ENTRY_SYNC_REQUEST:
                 log.info("Start Log Entry Sync, requestId={}", event.getEventID());
                 // Set the id of the event that caused the transition to the new state
                 // This is used to correlate trim or error events that derive from this state
                 LogReplicationState logEntrySyncState = fsm.getStates().get(LogReplicationStateType.IN_LOG_ENTRY_SYNC);
                 logEntrySyncState.setTransitionEventId(event.getEventID());
+                fsm.setBaseSnapshot(event.getMetadata().getLastTransferredBaseSnapshot());
```

Comment:
Why do there are two places to need to setup fsm.setBaseSnapshot, and fsm.setAckedTimestamp?

Reply:
Both are local variables in the FSM with different meaning. This is set when there is a transition from Initialized directly to Log Entry Sync, the first one indicates the latest base snapshot that was synced, the second the last log entry that was synced. And most likely if this is a continued log entry sync, these 2 values are different.

Reply:
I don't mean the two variables. I mean in the above case SNAPSHOT_TRANSFER_COMPLETE and the current case LOG_ENTRY_SYNC_REQUEST, they both setBaseSnapshot? Why do you need to setBaseSnapshot in two places?

Reply:
They are two different events which can happen independently. In the SNAPSHOT_TRANSFER_COMPLETE case, you're just starting and negotiation determines that the data is already transferred (therefore you need to reset/set the base snapshot to the one that represents the baseSnapshot of the ongoing snapshot sync). And in the second case, negotiation determined you can continue with log entry sync, you also initialize the values you just got to be aware of during negotiation.